pkg/envtest/crd.go

```diff
@@ -127,14 +127,17 @@ func WaitForCRDs(config *rest.Config, crds []client.Object, options CRDInstallOp
 		if err != nil {
 			return err
 		}
-		if crdVersion != "" {
-			gvs = append(gvs, schema.GroupVersion{Group: crdGroup, Version: crdVersion})
-		}
-
-		versions, _, err := unstructured.NestedSlice(crd.Object, "spec", "versions")
+		versions, found, err := unstructured.NestedSlice(crd.Object, "spec", "versions")
 		if err != nil {
 			return err
 		}
+
+		// gvs should be added here only if single version is found. If multiple version is found we will add those version
+		// based on the version is served or not.
+		if crdVersion != "" && !found {
```

Comment:
nit: (don't consider this blocking, but if for some reason you have to do another push, please fix): make `found` more descriptive, since it lasts more than a line.  For example, `isMultiVersion`